### ISSUE TITLE
Subclassing Builder with generics does not remove final modifier

### ISSUE THREAD

**Issue by sandorw** [immutables/immutables#387]
I'm trying to extend the immutable builder to hide the implementation like so:

```
@Value.Immutable
public abstract class TestCase<T> {

    public abstract T getValue();

    public static final class Builder<T> extends ImmutableTestCase.Builder<T> {}
}
```

Without generics, this works. With generics, the final modifier on the builder in the immutable implementation is not removed, so it won't compile (The type Builder cannot subclass the final class ImmutableTestCase.Builder<T>).

**Comment by elucash** [comment]
Thank you for reporting this! I've tried it with the version 2.2.8 and it works in Javac and Eclipse and there test cases for this to work. Is there can be anything specific regarding your setup? Can you try clean rebuild? What compiler do you use (build tool, IDE etc) so this is not compiling?

**Comment by sandorw** [comment]
I'm using eclipse and gradle. I bumped my dependency to 2.2.8 and got the case above working, but my actual use case (3 generic types) is still failing:

```
@Value.Immutable
public abstract class TestCase3<T, K, V> {

    public abstract T getValue();

    public static final class Builder<T, K, V> extends ImmutableTestCase3.Builder<T, K, V> {}
}
```

Thanks for the prompt support on this :)

**Comment by elucash** [comment]
There indeed some problem there, with 3 or 2 type params, investigating.

**Comment by elucash** [comment]
The fix released in 2.2.9 Thank you!
